android/src/app/organicmaps/util/Utils.java

```diff
@@ -810,4 +811,29 @@ public void onCompleted(final boolean success, @Nullable final String zipPath)
       });
     }
   }
+
+  @Nullable
+  public static <T> T getParcelable(Bundle args, String key, Class<T> clazz)
+  {
+    args.setClassLoader(clazz.getClassLoader());
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU)
+    {
+      return args.getParcelable(key, clazz);
+    }
+    return getParcelableOld(args, key, clazz);
+  }
+
+  @SuppressWarnings("deprecation")
+  @Nullable
+  private static <T> T getParcelableOld(android.os.Bundle args, String key, Class<T> clazz)
+  {
+    try
+    {
+      return clazz.cast(args.getParcelable(key));
+    }
+    catch (ClassCastException e)
+    {
+      return null;
```

Comment:
Why do you think that returning null is a better idea than just propagating the exception?

Reply:
The same logic as in `Bundle.getParcelable()`
And we have some places in code with null checks and some fields may accept null from parcelable

Reply:
It would be easier for us to debug if the ClassCastException will be thrown, no?

Reply:
Oh no, that looks over-engineered.
Why not to keep simple?
```
return args.getParcelable(key);
```
As it was before? We don't break any logic here and always ensure that passing T is _castable_

Reply:
@vng

Reply:
I understand, this is enough:
@SuppressWarnings("deprecation")
@Nullable
private static <T> T getParcelableOld(Bundle args, String key, Class<T> clazz)
{
  return args.getParcelable(key);
}
```
- we keep behaviour as-is (null return)
- args.getParcelable will print log itself, if any cast error. No need to print our custom logs